nixos/modules/services/networking/nbd.nix

```
@@ -0,0 +1,152 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nbd;
+  configFormat = pkgs.formats.ini { };
+  iniFields = with types; attrsOf (oneOf [ bool int float str ]);
+  serverConfig = configFormat.generate "nbd-server-config"
+    ({
+      generic = ({
+        user = "root";
+        group = "root";
+        port = cfg.server.listenPort;
+      }
+      // cfg.server.extraOptions
+      // (if (cfg.server.listenAddress != null)
+      then { listenaddr = cfg.server.listenAddress; }
+      else { }));
```

```suggestion
      // (optionalAttrs (cfg.server.listenAddress != null) {
        listenaddr = cfg.server.listenAddress;
      }));
```
also for other instances of this pattern.